src/voting/approval.ts

```diff
@@ -1,8 +1,24 @@
+function filterVotesWithInvalidChoice(votes, choices) {
+  return votes.filter((vote) => {
+    return (
+      Array.isArray(vote.choice) &&
+      // If choice index is not in choices, return false
+      vote.choice.every((choice) => choices?.[choice - 1] !== undefined) &&
+      // If any choice is duplicated, return false
+      vote.choice.length === new Set(vote.choice).size
+    );
+  });
+}
```

Comment:
@bonustrack @mktcode please let me know if you can think any cases that aren't covered here for approval voting.